### DISCUSSION TITLE
subscribe topic does not work when the client connects to an external broker

### DISCUSSION THREAD

**Discussion by Jsk1962** [hsaturn/TinyMqtt#92] [Q&A]
example:

  mqx_client.connect("192.168.1.2", 1883);  
  mqx_client.setCallback(mqtt_mesage_received2);   
  mqx_client.subscribe("topic_abc/#");    //  subscription is not working  and  mqx_client.mqx_2.connected()==false !!!

**Comment by Jsk1962** [comment]
mqx_client.subscribe("topic_abc/#"); // subscription is not working and mqx_client.connected()==false !!!

**Comment by hsaturn** [comment]
Hello, what is the broker you are using ? And is any other Mqtt client can connect to it ?

**Comment by Jsk1962** [comment]
Hi

I use RTU 950 (mqtt mosquitto).
This is strange because:
//--
mqx_client.connect("192.168.1.2", 1883);
mqx_client.setCallback(mqtt_mesage_received2);
//mqx_client.subscribe("topic_abc/#");
//---
When there is no subscribe to client works)
and when there is a call
mqx_client.subscribe("topic_abc/#");
it stops working.

Additional call
mqx_client.connect("192.168.1.2", 1883);
restores the connection but after publishing some information it disconnects.

Regards, Jurek

**Comment by Jsk1962** [comment]
I tested the sketch with ESP8266 uMQTTBroker
and everything works here.
